### PR TITLE
easyrpg-player: declare indirect deps with linkage

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
https://github.com/Homebrew/homebrew-core/actions/runs/10299733792/job/28508695864?pr=180477#step:4:1051
https://github.com/Homebrew/homebrew-core/actions/runs/10299733792/job/28508696529?pr=180477#step:3:76

### CODE REVIEW

File: Formula/e/easyrpg-player.rb [hunk 1]
```diff
@@ -35,6 +35,13 @@ class EasyrpgPlayer < Formula
   depends_on "sdl2"
   depends_on "speexdsp"
 
+  uses_from_macos "zlib"
+
+  on_macos do
+    depends_on "expat"
```
Comment (carlocab) [inline Formula/e/easyrpg-player.rb:41]:
```ruby
    uses_from_macos "expat"
```

Comment (chenrui333) [reply Formula/e/easyrpg-player.rb]: 👍

File: Formula/e/easyrpg-player.rb [hunk 2]
```diff
@@ -35,6 +35,13 @@ class EasyrpgPlayer < Formula
   depends_on "sdl2"
   depends_on "speexdsp"
 
+  uses_from_macos "expat"
```
Comment (carlocab) [inline Formula/e/easyrpg-player.rb:38]: Looks like this actually needs to be an unconditional dependency

Comment (chenrui333) [reply Formula/e/easyrpg-player.rb]: that actually makes sense.

File: Formula/e/easyrpg-player.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/e/easyrpg-player.rb]: why do we need linkage declaration for linux?

**Review by carlocab — APPROVED** [review]

**Pull request merged**